ttnn/cpp/ttnn/operations/moreh/moreh_dot_op/device/moreh_dot_program_factory.cpp

```diff
@@ -0,0 +1,166 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "moreh_dot_device_operation.hpp"
+#include "tt_dnn/op_library/moreh_helper_functions.hpp"
+#include "ttnn/deprecated/tt_dnn/op_library/work_split.hpp"
+#include "tt_metal/common/constants.hpp"
+#include "tt_metal/tt_metal.cpp"
+
+
+namespace ttnn::operations::moreh::moreh_dot_op {
+MorehDotOperation::SingleCore::cached_program_t MorehDotOperation::SingleCore::create(
+    const operation_attributes_t& operation_attributes,
+    const tensor_args_t& tensor_args,
+    tensor_return_value_t& tensor_return_value) {
+    using namespace tt;
+    using namespace tt::tt_metal;
+
+    const auto& input_tensor_a = tensor_args.input_tensor_a;
+    const auto& input_tensor_b = tensor_args.input_tensor_b;
+    auto& output_tensor = tensor_return_value;
+
+    auto src0_buffer = input_tensor_a.buffer();
+    auto src1_buffer = input_tensor_b.buffer();
+    auto dst_buffer = output_tensor.buffer();
+    float scaler = 1.0f;
+
+    tt::tt_metal::Program program{};
```

Comment:
```suggestion
   Program program{};
```
Since `using namespace tt::tt_metal` is declared at the top, I think it should be fine to make this change